pywbem_mock/_defaultinstanceprovider.py

```diff
@@ -0,0 +1,439 @@
+                _format("Cannot create instance because its creation "
+                        "class {0!A} does not exist in namespace {1!A}.",
+                        NewInstance.classname, namespace))
+
+        provider = self.get_registered_provider(namespace, 'instance',
+                                                NewInstance.classname)
+
+        if provider:
+            return provider.CreateInstance(namespace, NewInstance)
+
+        return self.default_instance_provider.CreateInstance(
+            namespace, NewInstance)
+
+    def DeleteInstance(self, InstanceName):
+        """
+        Dispatcher for the DeleteInstance method.
+
+        This method validates input parameters against the repository and
+        provider required types and  routes the method either to the default
+        provider or to a provider registered for the namespace and class in
+        InstanceName
+        """
+        provider = self.get_registered_provider(InstanceName.namespace,
+                                                'instance',
+                                                InstanceName.classname)
+
+        # Validate input parameters
+        namespace = InstanceName.namespace
+        self.validate_namespace(namespace)
+
+        # Test if corresponding class and instance already exists
+        if not self.class_exists(namespace, InstanceName.classname):
+            raise CIMError(
+                CIM_ERR_INVALID_CLASS,
+                _format("Class {0!A} in namespace {1!A} not found. "
+                        "Cannot delete instance {2!A}",
+                        InstanceName.classname, namespace, InstanceName))
+
+        instance_store = self.get_instance_store(namespace)
+        if not instance_store.exists(InstanceName):
+            raise CIMError(
+                CIM_ERR_NOT_FOUND,
+                _format("Instance {0!A} not found in CIM repository namespace "
+                        "{1!A}", InstanceName, namespace))
+
+        if provider:
+            return provider.DeleteInstance(InstanceName)
+
+        return self.default_instance_provider.DeleteInstance(
+            InstanceName)
+
+
+class DefaultInstanceProvider(BaseProvider):
+    """
+    This class defined those instance provider methods that may have user-
+    defined providers that override the default provider implementation in this
+    class.
+
+    User providers are defined by creating a subclass of this class and
+    defining a new provider method for one of the methods in this class with
+    the same signature.
+
+    Note that user-defined providers may, in turn, call the default providers
+    in this class.
+
+    """
+    def __init__(self, conn):
+        """
+        """
+
+        # TODO: This not correct way to make cimrepository available to
+        # multiple subclasses
+        super(DefaultInstanceProvider, self).__init__(
+            conn=conn, cimrepository=None)
+
+        self.cimrepository = conn.cimrepository
```

Comment:
Instead of passing in `conn` and then using exactly one attribute from it, I suggest to pass in that one attribute. That avoids the back reference and makes it more clear what information is actually passed.

Reply:
**DONE**